### PR TITLE
enh(gorgone) export conf for central

### PR DESCRIPTION
**Pull request by loiclau**
## Description

action button for central

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x
- [ ] 18.10.x
- [ ] 19.04.x
- [ ] 19.10.x
- [x] 20.04.x (master)

<h2> How this pull request can be tested ? </h2>

go to Configuration  >  Pollers  with gorgone_protocol in zmq for central

## Checklist

- [ ] I followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/include/configuration/configServers/popup/popup.php [hunk 1]
```diff
@@ -83,93 +82,109 @@
...
-    $gorgoneError = true;
-    $dataError = $ex->getMessage();
-}
-
-if (!empty($dataError)) {
-    $config = $dataError;
-} elseif (in_array($server['ns_ip_address'], $remotesServerIPs)) {
-    //config for remote
-    $config = file_get_contents('./remote.yaml');
+if ($server['localhost'] == '1') {
```
Comment (kduret) [inline www/include/configuration/configServers/popup/popup.php:86]:
```suggestion
if ($server['localhost'] === '1') {
```

File: www/include/configuration/configServers/popup/central.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    - name: nodes
+      package: "gorgone::modules::centreon::nodes::hooks"
+      enable: true
+    - name: proxy
+      package: "gorgone::modules::core::proxy::hooks"
+      enable: true
+    - name: legacycmd
+      package: "gorgone::modules::centreon::legacycmd::hooks"
+      enable: true
+      cmd_dir: "/var/lib/centreon/centcore/"
```
Comment (sc979) [inline www/include/configuration/configServers/popup/central.yaml:36]: shouldn't these two path be retrieved from current configuration ? ie : source installation

File: www/include/configuration/configServers/popup/central.yaml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      cache_dir_trap: "/etc/snmp/centreon_traps"
+      remote_dir: "/var/cache/centreon/config/remote-data/"
+    - name: engine
+      package: "gorgone::modules::centreon::engine::hooks"
+      enable: true
+      command_file: "__COMMAND__"
+    - name: statistics
+      package: "gorgone::modules::centreon::statistics::hooks"
+      enable: true
+      broker_cache_dir: "/var/cache/centreon/broker-stats/"
```
Comment (sc979) [inline www/include/configuration/configServers/popup/central.yaml:48]: shouldn't next path values be retrieved from current configuration ? ie : source installation

**Comment by loiclau** [comment]
> I think it does not work for a remote server, cause it's handled inside the if which check localhost property

page does not exist in remote.

**Comment by kduret** [comment]
> > I think it does not work for a remote server, cause it's handled inside the if which check localhost property
> 
> page does not exist in remote.

Yes, but you can have the popup of the remote server configuration from central UI, right ?

**Comment by loiclau** [comment]
> > > I think it does not work for a remote server, cause it's handled inside the if which check localhost property
> > 
> > 
> > page does not exist in remote.
> 
> Yes, but you can have the popup of the remote server configuration from central UI, right ?

remote and poller are manage on the else

File: www/include/configuration/configServers/popup/popup.php [hunk 4]
```diff
@@ -83,93 +82,117 @@
...
+            if ($timeout > 30) {
+                // add 10 s for the next server
+                $timeout -= 10;
+                $gorgoneError = true;
+                $dataError .= '
+      - error : TimeOut error for poller ' . $serverId . ' We can\'t get log';
+                continue;
+            }
+
+            $thummprintReponse = json_decode($lastActionLog->getData(), true);
```
Comment (jeremyjaouen) [inline www/include/configuration/configServers/popup/popup.php:134]:
```PHP
$thumbprintReponse = json_decode($lastActionLog->getData(), true);
```

File: www/include/configuration/configServers/popup/popup.php [hunk 5]
```diff
@@ -83,93 +82,117 @@
...
+            if ($lastActionLog->getCode() === \Centreon\Domain\Gorgone\Response::STATUS_OK) {
+                $thumbprints .= '
+      - key: ' . $thummprintReponse['data']['thumbprint'];
```
Comment (jeremyjaouen) [inline www/include/configuration/configServers/popup/popup.php:137]:
```PHP
- key: ' . $thumbprintReponse['data']['thumbprint'];
```

File: www/include/configuration/configServers/popup/popup.php [hunk 6]
```diff
@@ -83,93 +82,117 @@
...
+            } else {
+                $gorgoneError = true;
+                $dataError .= '
+      - error : Poller ' . $serverId . ' : ' . $thummprintReponse['message'];
```
Comment (jeremyjaouen) [inline www/include/configuration/configServers/popup/popup.php:141]:
```PHP
 - error : Poller ' . $serverId . ' : ' . $thumbprintReponse['message'];
```

File: www/include/configuration/configServers/popup/popup.php (hunk 4 above)
Comment (sc979) [reply www/include/configuration/configServers/popup/popup.php:134]: + camelCase ?

Comment (jeremyjaouen) [reply www/include/configuration/configServers/popup/popup.php:134]: i don't think camelCase is mandatory here as thumbprint is a single word

Comment (sc979) [reply www/include/configuration/configServers/popup/popup.php:134]: a "s" is missing too ;)

Comment (sc979) [inline www/include/configuration/configServers/popup/popup.php:134]:
```suggestion
            $thumbprintResponse = json_decode($lastActionLog->getData(), true);
```

File: www/include/configuration/configServers/popup/popup.php (hunk 5 above)
Comment (sc979) [inline www/include/configuration/configServers/popup/popup.php:137]:
```suggestion
      - key: ' . $thumbprintResponse['data']['thumbprint'];
```

File: www/include/configuration/configServers/popup/popup.php (hunk 6 above)
Comment (sc979) [inline www/include/configuration/configServers/popup/popup.php:141]:
```suggestion
      - error : Poller ' . $serverId . ' : ' . $thumbprintResponse['message'];
```

**Pull request merged**